src/core/layout/qgslayoutitemlegend.cpp

```diff
@@ -680,6 +683,11 @@ void QgsLayoutItemLegend::setLinkedMap( QgsLayoutItemMap *map )
   }
 
   updateFilterByMap();
+
+  // unsure if needed
+  QgsExpressionContext context = updateExpressionContext();
+  mLegendModel->setLayoutExpContext( &mExpContext );
```

Comment:
I am adding these as a safety measure to ensure that the member mExpContext is kept up to date, is this a good thing to update for every new linked map or should I not do that.